### ISSUE TITLE
Macro-op-fusion: Pattern design of ADC and SBB

### ISSUE THREAD

**Issue by mohanson** [nervosnetwork/ckb-vm#169]
# ADC

Adding a0, a1, and a2 with results in a0 and carry-out in a1.

```
add  a0, a0, a1
sltu a1, a0, a1
add  a0, a0, a2
sltu a2, a0, a2
or   a1, a1, a2
```

- a0 ~ a2 are used as placeholders
- a0, a1, a2 != ZERO

# SBB

Do a0 - a1 - a2 with results in a0 and carry-out in a1.

```
sub  a1, a0, a1
sltu a3, a0, a1
sub  a0, a1, a2
sltu a2, a1, a0
or   a1, a2, a3
```

- a0 ~ a3 are used as placeholders
- a3 is a temporary register
- a0, a1, a2, a3 != ZERO

We need to read 5 instructions to really determine whether these 5 instructions can be fused into a mop instruction, but in fact, when the first instruction is read, we have a high accuracy rate to judge whether the first instruction is the beginning of an sbb mop instructions. This way `sub a1, a0, a1` of writing `sub` is not very common, in most cases we will write like this `sub a0, a0, a1`. Therefore, this effectively distinguishes `sbb` and normal `sub`.

Test code: https://github.com/mohanson/ckb-vm/tree/adcsbb

**Comment by mohanson** [comment]
cc @XuJiandong @xxuejie

**Comment by mohanson** [comment]
Features have been merged by #170
